src/Nova/Booking.php

```diff
@@ -0,0 +1,139 @@
+                return $this->slot->formatted_start;
+            })->sortable(),
+            Date::make('Game Date', 'slot.date')->sortable(),
+            Number::make('Participants'),
+            DateTime::make('Purchased', 'created_at')->sortable(),
+        ]);
+    }
+
+    public function fields(Request $request)
+    {
+        return array_filter([
+            nova('slot') ? BelongsTo::make('Slot', 'slot', nova('slot')) : null,
+            Text::make('Game Start', 'slot.start_at', function () {
+                return $this->slot->formatted_start;
+            })->exceptOnForms(),
+            Number::make('Participants')->exceptOnForms(),
+            Boolean::make('Private Game?', 'is_private')->exceptOnForms(),
+
+            new Panel('Customer', $this->customerFields()),
+            new Panel('Cost', $this->costFields()),
+
+            nova('signature') ? HasMany::make('Signatures', 'signatures', nova('signature')) : null,
+            nova('note') ? MorphMany::make('Notes', 'notes', nova('note')) : null,
+
+            new Panel('Data Fields', $this->dataFields()),
+        ]);
+    }
+
+    protected function customerFields()
+    {
+        return [
+            Text::make('Name', 'order.customer.user.full_name')->exceptOnForms(),
+            PhoneNumber::make('Phone', 'order.customer.phone_number')->format('###-###-####')->disableValidation()->useMaskPlaceholder()->linkOnDetail()->exceptOnForms(),
+            Email::make('Email', 'order.customer.user.email')->clickable()->exceptOnForms(),
+            Text::make('Address', 'order.customer.address')->exceptOnForms(),
+            Text::make('City', 'order.customer.city')->exceptOnForms(),
+            Text::make('State', 'order.customer.state')->exceptOnForms(),
+            Number::make('Zip', 'order.customer.zip')->exceptOnForms(),
+        ];
+    }
+
+    protected function costFields()
+    {
+        return array_filter([
+            nova('order') ? BelongsTo::make('Order', 'order', nova('order'))->hideWhenUpdating()->exceptOnForms() : null,
+            nova('rate') ? BelongsTo::make('Rate', 'rate', nova('rate'))->nullable()->exceptOnForms() : null,
+            nova('tax') ? BelongsTo::make('Tax', 'tax', nova('tax'))->nullable()->exceptOnForms() : null,
+            nova('fee') ? BelongsTo::make('Fee', 'fee', nova('fee'))->nullable()->exceptOnForms() : null,
+            Currency::make('Amount')->asMinorUnits()->exceptOnForms(),
+            Currency::make('Taxes', 'total_taxes')->asMinorUnits()->exceptOnForms(),
+            Currency::make('Fees', 'total_fees')->asMinorUnits()->exceptOnForms(),
+        ]);
+    }
+
+    protected function dataFields(): array
+    {
+        return [
+            ID::make(),
+            DateTime::make('Created At')->exceptOnForms(),
+            DateTime::make('Updated At')->exceptOnForms(),
+        ];
+    }
+
+    public function filters(Request $request)
+    {
+        return [
+            new Filters\RealizedBookings,
```

Comment:
@drewroberts Can you tell me what packages contain these filters so I can refactor them in the new way.

Reply:
You'll see the filters in the Nova resources on the main repository. We don't need to add any new ones at this point.